packages/vaadin-grid/src/vaadin-grid-row-details-mixin.js

```diff
@@ -62,6 +62,20 @@ export const RowDetailsMixin = (superClass) =>
       ];
     }
 
+    ready() {
+      super.ready();
+
+      this._detailsCellResizeObserver = new ResizeObserver((entries) => {
+        entries.forEach(({ target: cell }) => {
+          cell.parentElement.style.setProperty('padding-bottom', `${cell.offsetHeight}px`);
+        });
+
+        // This workaround is needed until Safari also supports
+        // ResizeObserver.observe with {box: 'border-box'}
+        this.__virtualizer.__adapter._resizeHandler();
```

Comment:
ResizeObserver observes content-box by default so the padding applied to the row elements above would get ignored by the virtualized ResizeObserver. Needed to add this temporary, manual _resizehandler invocation via private APIs :/ Once ResizeObserver supports border-box on all supported browsers, we can make virtualized use it and remove this line.